src/core/net/dnssd_server.hpp

```diff
@@ -403,6 +420,8 @@ class Server : public InstanceLocator, private NonCopyable
     otDnssdQuerySubscribeCallback   mQuerySubscribe;
     otDnssdQueryUnsubscribeCallback mQueryUnsubscribe;
     TimerMilli                      mTimer;
+
+    Counters mCounters;
```

Comment:
Do we need to initialize `mCounters`?

Reply:
Good catch! I just updated the constructor of `Server`.

Reply:
@superwhd I noticed you added `mCounters()` in `Server` constructor with the goal to have `mCounters `struct be initialized as zero.

While this should work in most cases (and I believe maybe even mandated by C++11 onwards), I suggest we do the clearing explicitly.

We can define
```
class Counters : public otDnssdCounters, public Clearable<Counter> 
{ 
};
``` 
Then we can use `Clear()` method (or add a constructor that does clear the counters explicitly).

The reason I am suggesting this is that I have seen some C++ toolchains (specially custom embedded tool chains) that may not do the zero-initialization properly (or may allow it to be disabled through certain config).

In general, it is safer to do this explicitly in OT code (it does not add any overhead and we we avoid weird behaviors/bugs due to OT being build on specific toolchains or under certain configs).